cmd/prometheus/main.go

```diff
@@ -467,6 +467,9 @@ func main() {
 	serverOnlyFlag(a, "storage.tsdb.delayed-compaction.max-percent", "Sets the upper limit for the random compaction delay, specified as a percentage of the head chunk range. 100 means the compaction can be delayed by up to the entire head chunk range. Only effective when the delayed-compaction feature flag is enabled.").
 		Default("10").Hidden().IntVar(&cfg.tsdb.CompactionDelayMaxPercent)
 
+	serverOnlyFlag(a, "storage.tsdb.cache-all-symbols-for-compaction", "Enable caching of all symbols for compaction.").
+		Default("true").Hidden().BoolVar(&cfg.tsdb.CacheAllSymbols)
```

Comment:
I still question the need to make this CLI-controllable: no Prometheus user has reported an issue.  Making it configurable in code is all you need for the problem you had, and would simplify this PR a bit.

Reply:
I removed the flag. PTAL.